tests/TestUtils/MockTokenAuth.cs

```diff
@@ -0,0 +1,116 @@
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using Microsoft.IdentityModel.Tokens;
+
+namespace TestUtils
+{
+    public static class MockTokenAuth
+    {
+        /// <summary>
+        /// Generate a mock authorization bearer token using standard JWT libraries
+        /// </summary>
+        /// <param name="audience">Token audience (mandatory)</param>
+        /// <param name="expiresIn">Token expiration in seconds (default: 3600)</param>
+        /// <param name="issuer">Token issuer (default: "test-issuer")</param>
+        /// <param name="keyId">Key ID for the token header (default: "test-key-id")</param>
+        /// <param name="additionalClaims">Additional claims to be added to the token</param>
+        /// <returns>Token information including the bearer token string</returns>
+        public static TokenInfo GenerateToken(
+            string audience = "test-client-id",
+            int expiresIn = 3600,
+            string issuer = "test-issuer",
+            string keyId = "test-key-id",
+            IEnumerable<Claim>? additionalClaims = null)
+        {
+            // Create a symmetric security key with a consistent key for testing
+            // In production, you'd use a secure key management system
+            byte[] keyBytes = new byte[32]; // 256 bits for HMAC SHA-256
+            using (var rng = RandomNumberGenerator.Create())
+            {
+                rng.GetBytes(keyBytes);
+            }
+            var securityKey = new SymmetricSecurityKey(keyBytes);
+            securityKey.KeyId = keyId; // Set the kid explicitly on the security key
+
+            // Create signing credentials with key ID
+            var signingCredentials = new SigningCredentials(securityKey, SecurityAlgorithms.HmacSha256);
+
+            // Set token expiration
+            var issuedAt = DateTime.UtcNow;
+            var expiresAt = issuedAt.AddSeconds(expiresIn);
+
+            // Create base claims for the token
+            var claims = new List<Claim>
+            {
+                new Claim(JwtRegisteredClaimNames.Iss, issuer),
+                new Claim(JwtRegisteredClaimNames.Aud, audience),
+                new Claim(JwtRegisteredClaimNames.Iat, EpochTime.GetIntDate(issuedAt).ToString(), ClaimValueTypes.Integer64)
+            };
+
+            // Add any additional claims provided
+            if (additionalClaims != null)
```

Comment:
there are a couple of checks of additionalClaims' nullability.
Also we risk multiple enumerations of an enumerable, which might cause SonarCube to grouse.

Suggest doing the null check once, and converting the IEnumerable to a List:

```
            // Add any additional claims provided
            var additionalClaimsList = additionalClaims == null ? new List<Claim>() : new List<Claim>(additionalClaims);
            claims.AddRange(additionalClaimsList);
```